File: eBarbershop.Model/NarudzbaProizvodi.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eBarbershop.Model
{
    public class NarudzbaProizvodi
    {
        public int NarudzbaProizvodiId { get; set; }

        public int NarudzbaId { get; set; }

        public int ProizvodId { get; set; }
        public virtual Proizvod Proizvod { get; set; }

        public int Kolicina { get; set; }
    }

}

File: UI/ebarbershop_mobile/lib/models/narudzbaproizvodi.dart
import 'package:ebarbershop_mobile/models/product.dart';
import 'package:json_annotation/json_annotation.dart';

part 'narudzbaproizvodi.g.dart';

@JsonSerializable()
class NarudzbaProizvodi {
  int? narudzbaProizvodiId;
  int? narudzbaId;
  int? proizvodId;
  Product? proizvod;
  int? kolicina;
  NarudzbaProizvodi(this.narudzbaProizvodiId, this.narudzbaId, this.proizvodId,
      this.proizvod, this.kolicina);

  factory NarudzbaProizvodi.fromJson(Map<String, dynamic> json) =>
      _$NarudzbaProizvodiFromJson(json);

  Map<String, dynamic> toJson() => _$NarudzbaProizvodiToJson(this);
}
